### DISCUSSION TITLE
implementation of XState in the AssemblyScript language

### DISCUSSION THREAD

**Discussion by edbek** [statelyai/xstate#2934] [Ideas]
Can we expect the implementation of XState in the AssemblyScript language?

**Comment by Andarist** [comment]
There is no plan to work on implementations in other languages right now - but it's something that we are keeping an eye on and it's something that we are interested in pursuing in the future. Our team is small now though and we have to pick our battles - so, at the time being, it means that our focus is on the JS/TS.

**Reply by edbek** [reply]
I would like to draw your attention to the fact that there is an urgent need for support for WASM. But unfortunately, full-fledged compilation in WASM is possible in AssemblyScript, C, C ++, Go languages. Thus, in the future it would be interesting to implement XState in the Go or AssemblyScript languages.
Thanks for your work.

**Reply by Andarist** [reply]
> Thus, in the future it would be interesting to implement XState in the Go or AssemblyScript languages.

We totally agree that this has huge potential - it's just like I've mentioned: at the moment the team is small and we just can't "afford" porting XState to other languages. At the moment we are focused on building visual editing tools and some of that work (and its reception) will help us to prioritize further steps.

**Comment by drush** [comment]
I would add that just getting `xstate-fsm` working is probably sufficient for most needs.

**Comment by drush** [comment]
Another thread on the same topic https://github.com/statelyai/xstate/discussions/2761
